### DISCUSSION TITLE
compile time

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#11790] [Q&A]
Hi,

on my system it takes aeons to compile even Avalonia template. Is this normal?

My System:
OS: Ubuntu 23.04 with KDE
CPU: Ryzen 7 5700X (8 cores)
RAM: 64GB

should be enough power.

A console application is compiled withing less than a second

**Comment by drei97** [comment]
... this is only with 11.0 not with 0.10.xx

**Reply by timunie** [reply]
which version of 11.0? Please be precise here as 11.0 preview8 is different to RC1 is different to lastest nightly...

**Comment by maxkatz6** [comment]
Please run "dotnet build /bl" and check your msbuild log file. You can open it with https://msbuildlog.com/
It will show what tasks take the most of build time.

**Comment by HobbesKo** [comment]
I have a similar issue.
My project used to build in less than 5 seconds (using JetBrains Rider) before I went to Preview 8. With Preview 8 the build took 20 seconds plus.
I cured the slow build by reverting to Preview 6. Once again the build time went back to 5 seconds.

Now, with RC1, the build time is again at least 20 seconds.

**Comment by workgroupengineering** [comment]
It is probably the same as issue #11653. Try last night's build.

**Comment by HobbesKo** [comment]
Using last night's build has made no difference to my project's build speed, it's still really slow.

Running `dotnet build` takes 3 seconds. If I then go to Rider and build, that also takes about 3 seconds. If I close and re-open Rider, the build time goes back to 20 seconds.
I originally suspected Rider, but the fact that the build time in Preview 6 was fast, I no longer  think that is the case.
 
The telemetry state appears to make no difference, I've tried both enabled and disabled.

**Comment by drei97** [comment]
here my binlog

![2023-06-15 09_44_15-C__data_msbuild binlog - MSBuild Structured Log Viewer](https://github.com/AvaloniaUI/Avalonia/assets/68755045/302ec31a-efc6-4c2f-97d0-b0d0e704522d)

and 

![2023-06-15 09_43_02-C__data_msbuild binlog - MSBuild Structured Log Viewer](https://github.com/AvaloniaUI/Avalonia/assets/68755045/73b1d023-8321-4ef7-9f03-fc8005646c68)

**Reply by workgroupengineering** [reply]
Please compare rc1 binlog with binlogo to Preview 6.

**Reply by drei97** [reply]
how to create a test with preview 6? On console I only have a choice of templates 0.10.21 and 11.0.0-rc1.1

**Reply by workgroupengineering** [reply]
- create app 11.0.0-rc1.1 or using your app 11.0.0-rc1.1,
- edit your csproject and change PackageReference Version of all Alavalonia Package to 11.0.0-Preview6
- clean
- build with binlog

**Reply by maxkatz6** [reply]
Doesn't look like it's anything related to Avalonia so far.
Even though it's hard to tell why _copyfilesmarkedcopylocal task is slow here.

**Reply by maxkatz6** [reply]
In the same tool you can double click on this _copyfilesmarkedcopylocal task and see what parameters/items it has on input. Possibly there is an enormous amount of files to be copied.

**Comment by drei97** [comment]
After several runs I can say:

it is NOT the compiling which takes that long, it is the time needed to bring the Mainwindow on screen.

I have crosschecked the logs created by

```
time dotnet clean && time dotnet build -bl)
```

and the results are:
cleaning takes around 1sec and building around 10sec to 11sec and there are no differences between Prev6 and RC1.1!

But if I press "RUN" in the IDE (Rider 2023.1) Prev6 is coming up instantaneously and RC1.1 takes its time.

**Reply by drei97** [reply]
This is for OS Kubuntu 23.04 and IDE Rider 2023.1.

Just made a test with OS Win11 and IDE VS 2022. The start up of the RC1.1 Mainwindow is much faster as with Linux

**Reply by HobbesKo** [reply]
> But if I press "RUN" in the IDE (Rider 2023.1) Prev6 is coming up instantaneously and RC1.1 takes its time.

That's exactly the issue that I have. I've tried different versions of Rider but that has made no difference. (Win11)

**Reply by drei97** [reply]
Now I also tried to start from VS, start from Rider and start from Console.

It is a Rider issue! VS and Console starts the App within 1.2 sec and Rider takes 31sec

**Reply by drei97** [reply]
1sec - 2sec and not 1.2secs !!

**Reply by timunie** [reply]
Please compare to nightly builds, not to rc1. Also maybe file an issue to Rider with sample and binlog attached.

**Comment by HobbesKo** [comment]
11.0.0-rc2.2 seems to have fixed my compile time issue.
